### PR TITLE
🎊  Copilot Chat

### PR DESCRIPTION
**Pull request by TerminalFi**
Current status

TerminalFi 2024-07-02 at 15 29 07@2x

### CODE REVIEW

File: plugin/client.py [hunk 1]
```diff
@@ -197,7 +198,7 @@ def version() -> str:
     def editor_info(cls) -> dict[str, Any]:
         return {
             "editorInfo": {
-                "name": "Sublime Text",
+                "name": "vscode",
```
Comment (jfcherng) [inline plugin/client.py:201]: Fwiw, this seems to be a must.

**Comment by jfcherng** [comment]
About handling `$/progress`, https://github.com/sublimelsp/LSP/issues/2494

**Comment by jfcherng** [comment]
image

**Comment by jfcherng** [comment]
It also keep auto filling chat question for me 🤔

**Comment by TerminalFi** [comment]
1. That's copilot suggested follow up. Probably need different approach 😀 
2. Yea, will have to check how vscode does it

**Comment by TerminalFi** [comment]
> image

I think these are both fixed in latest

**Comment by TerminalFi** [comment]
I'm still working on the other two

1. Auto fill with suggested next question
2. What to do when asking while waiting

**Comment by TerminalFi** [comment]
> I'm still working on the other two
> 
> 1. Auto fill with suggested next question
> 2. What to do when asking while waiting

1. Follow up is a link at the base of the chat panel
2. We now prevent user submitting a new question while awaiting the next question. This is inline with VS Code

**Comment by TerminalFi** [comment]
The refactor is beautiful

**Pull request merged**